pkg/sql/drop_view.go

```diff
@@ -236,12 +236,15 @@ func (p *planner) dropViewImpl(
 			if err != nil {
 				return cascadeDroppedViews, err
 			}
-
-			cascadedViews, err := p.dropViewImpl(ctx, dependentDesc, queueJob, "dropping dependent view", behavior)
-			if err != nil {
-				return cascadeDroppedViews, err
+			// Check if the dependency was already marked as dropped,
+			// while dealing with any earlier dependent views.
+			if !dependentDesc.Dropped() {
+				cascadedViews, err := p.dropViewImpl(ctx, dependentDesc, queueJob, "dropping dependent view", behavior)
+				if err != nil {
+					return cascadeDroppedViews, err
+				}
+				cascadeDroppedViews = append(cascadeDroppedViews, cascadedViews...)
 			}
-			cascadeDroppedViews = append(cascadeDroppedViews, cascadedViews...)
 			cascadeDroppedViews = append(cascadeDroppedViews, qualifiedView.FQString())
```

Comment:
Do we still want to append it here if it's already been dropped? Can you augment the test to look at the eventlog event and see what names it has?